proto/sifnode/dispensation/v1/types.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+package sifnode.dispensation.v1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/coin.proto";
+
+option go_package = "github.com/Sifchain/sifnode/x/dispensation/types";
+
+message GenesisState {
+  repeated string address_whitelist = 1;
+  string          admin_address     = 2;
+}
+
+// Distribution type enum
+enum DistributionType {
+  // Unspecified distribution type
+  DISTRIBUTION_TYPE_UNSPECIFIED = 0;
+  // Airdrop distribution type
+  DISTRIBUTION_TYPE_AIRDROP = 1;
+}
+
+message DistributionRecord {
+  string claim_status                  = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Uint",
+    (gogoproto.nullable)   = false,
+    (gogoproto.moretags)   = "yaml:\"claim_status\""
+  ];
+  string distribution_name             = 2;
+  string recipient_address             = 3;
+  repeated cosmos.base.v1beta1.Coin coins = 4 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (gogoproto.moretags)     = "yaml:\"coins\""
+  ];
+  string distribution_start_height     = 5 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false,
+    (gogoproto.moretags)   = "yaml:\"distribution_start_height\""
+  ];
+  string distribution_completed_height = 6 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false,
+    (gogoproto.moretags)   = "yaml:\"distribution_completed_height\""
+  ];
+}
+
+message DistributionRecords {
+  repeated DistributionRecord distribution_records = 1;
+}
+
+message DistributionRecordsResponse {
+  DistributionRecords distribution_records = 1;
+  string height = 2 [
```

Comment:
why not make this int64 or uint64?

Reply:
I think int64 might be apt , becuase we set height as -1 for new records